x/rewards/keeper/providers_test.go

```diff
@@ -0,0 +1,420 @@
+	_, err = ts.TxSubscriptionBuy(consumerAcc.Addr.String(), consumerAcc.Addr.String(), ts.plan.Index, 1, false)
+	require.Nil(t, err)
+
+	msg := ts.SendRelay(providerAcc1.Addr.String(), consumerAcc, []string{ts.spec.Index}, ts.plan.Price.Amount.Uint64()/2)
+	_, err = ts.TxPairingRelayPayment(msg.Creator, msg.Relays...)
+	require.Nil(t, err)
+
+	msg = ts.SendRelay(providerAcc2.Addr.String(), consumerAcc, []string{ts.spec.Index}, ts.plan.Price.Amount.Uint64())
+	_, err = ts.TxPairingRelayPayment(msg.Creator, msg.Relays...)
+	require.Nil(t, err)
+
+	msg = ts.SendRelay(providerAcc3.Addr.String(), consumerAcc, []string{ts.spec.Index}, ts.plan.Price.Amount.Uint64()*2)
+	_, err = ts.TxPairingRelayPayment(msg.Creator, msg.Relays...)
+	require.Nil(t, err)
+
+	// first months there are no bonus rewards, just payment ffrom the subscription
+	ts.AdvanceMonths(1)
+	ts.AdvanceBlocks(ts.BlocksToSave() + 1)
+
+	res, err := ts.QueryDualstakingDelegatorRewards(providerAcc1.Addr.String(), "", "")
+	require.Nil(t, err)
+	require.Len(t, res.Rewards, 1)
+	_, err = ts.TxDualstakingClaimRewards(providerAcc1.Addr.String(), "")
+	require.Nil(t, err)
+
+	res, err = ts.QueryDualstakingDelegatorRewards(providerAcc2.Addr.String(), "", "")
+	require.Nil(t, err)
+	require.Len(t, res.Rewards, 1)
+	_, err = ts.TxDualstakingClaimRewards(providerAcc2.Addr.String(), "")
+	require.Nil(t, err)
+
+	res, err = ts.QueryDualstakingDelegatorRewards(providerAcc3.Addr.String(), "", "")
+	require.Nil(t, err)
+	require.Len(t, res.Rewards, 1)
+	_, err = ts.TxDualstakingClaimRewards(providerAcc3.Addr.String(), "")
+	require.Nil(t, err)
+
+	// now the provider should get all of the provider allocation
+	ts.AdvanceMonths(1)
+	distBalance := ts.Keepers.Rewards.TotalPoolTokens(ts.Ctx, types.ProviderRewardsDistributionPool)
+	ts.AdvanceEpoch()
+
+	res1, err := ts.QueryDualstakingDelegatorRewards(providerAcc1.Addr.String(), "", "")
+	require.Nil(t, err)
+	require.Len(t, res.Rewards, 1)
+	// we sub 3 because of truncating
+	require.Equal(t, res1.Rewards[0].Amount.Amount, distBalance.QuoRaw(7).SubRaw(3))
+	_, err = ts.TxDualstakingClaimRewards(providerAcc1.Addr.String(), providerAcc1.Addr.String())
+	require.Nil(t, err)
+
+	res2, err := ts.QueryDualstakingDelegatorRewards(providerAcc2.Addr.String(), "", "")
+	require.Nil(t, err)
+	require.Len(t, res.Rewards, 1)
+	// we sub 1 because of truncating
+	require.Equal(t, res2.Rewards[0].Amount.Amount, distBalance.QuoRaw(7).MulRaw(2))
+	_, err = ts.TxDualstakingClaimRewards(providerAcc2.Addr.String(), providerAcc2.Addr.String())
+	require.Nil(t, err)
+
+	res3, err := ts.QueryDualstakingDelegatorRewards(providerAcc3.Addr.String(), "", "")
+	require.Nil(t, err)
+	require.Len(t, res.Rewards, 1)
+	// we add 6 because of truncating
+	require.Equal(t, res3.Rewards[0].Amount.Amount, distBalance.QuoRaw(7).MulRaw(4).AddRaw(6))
+	_, err = ts.TxDualstakingClaimRewards(providerAcc3.Addr.String(), providerAcc3.Addr.String())
+	require.Nil(t, err)
+}
```

Comment:
can we do a test that passes 49 months? so the boos runs out i want to see this edge case doesn't panic